WalletWasabi/BitcoinCore/CachedRpcClient.cs

```diff
@@ -0,0 +1,209 @@
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Primitives;
+using NBitcoin;
+using NBitcoin.RPC;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.BitcoinCore
+{
+	public class CachedRpcClient : RpcClientBase
+	{
+		private CancellationTokenSource _tipChangeCancellationTokenSource;
+
+		public CachedRpcClient(RPCClient rpc, IMemoryCache cache)
```

Comment:
Shouldn't this take IRPCClient?

Reply:
I cannot do it because the `CachedRpcClient` only can cache a few methods. For that reason it inherits from `RpcClientBase` that it the class that wraps the real NBitcoin `RPCClient`. 

In order to make it work with an `IRpcClient` I should need to break the inheritance link and make the `CachedRpcClient` to implement (simply pass-through) all the `IRpcClient` methods.

I am not against that. I did it in this way to by-pass some way the self-imposed restriction on the size of the PR (this is a big PR already).

Reply:
I got it! ACK.